### DISCUSSION TITLE
Is it possible to create strongly typed form fragment with ability to map the values to a key on a more complex form?

### DISCUSSION THREAD

**Discussion by jdpt0** [react-hook-form/react-hook-form#8730] [Q&A] [answered]
For example, if I had an address object:
```javascript
{
  "name": string,
  "street" string
}
```

Which I will collect the data for by using this fragment:

```javascript
const AddressFormFragment = ({ register }: { register: UseFormRegister<FieldValues> }) => {
  return <>
    <input
      {...register("name", {
          required: true,
      })}
      placeholder={"Name"}
    />
    <input
      {...register("street", {
        required: true,
      })}
      placeholder={"Street"}
    />
  </>
}
```

If I wanted to use this fragment to collect data on two different keys of a parent form:

```javascript
{
   "deliveryAddress": {
    "name": string,
    "street": string,
  },
  "invoiceAddress": {
    "name": string,
   "street": string
  }
}
```

How could I go about this without having to copy and paste the input fields with `deliveryAddress.name`, `invoiceAddress.name` etc as names for the inputs while maintaining strong typescript types?

I tried this but get this error:

![image](https://user-images.githubusercontent.com/19761394/180875824-6c5b7747-cdf7-401a-a244-2a240b7066c3.png)

**Comment by Moshyfawn** [comment, marked as answer]
Take a look at [this RFC](https://github.com/react-hook-form/react-hook-form/discussions/7354). Does it look like something you're trying to achieve?

**Reply by jdpt0** [reply]
Yes, this is exactly the problem I'm wanting to solve, thank you! Is this already in V8 then?

**Reply by Moshyfawn** [reply]
You can already try out v8 **alpha** by running `npm i react-hook-form@beta`, but the team is still working on bringing v8 to life, currently without an ETA.

Here's [a codesandbox example](https://codesandbox.io/s/falling-cookies-s1xzb?file=%2Fsrc%2FApp.tsx)
